g2p_registry_rest_api/models/individual.py

```diff
@@ -46,3 +48,19 @@ def validate_family_name(cls, v):  # noqa: B902
                 error_description="Family name is mandatory",
             )
         return v
+
+    @validator("gender")
+    def validate_gender(cls, value):
+        gender_type = request.env["ir.config_parameter"].get_param(
+            "g2p_registry.gender_config"
+        )
+        gender_type = [item.strip() for item in gender_type.split(",")]
+        if value and value not in gender_type:
+            raise G2PApiValidationError(
+                error_message=G2PErrorCodes.G2P_REQ_008.get_error_message(),
+                error_code=G2PErrorCodes.G2P_REQ_008.get_error_code(),
+                error_description=(
+                    "Invalid Gender - %s gender should be - %s" % (value, gender_type)
+                ),
```

Comment:
Similar parsing of map/json/selection array elements might be required here. According to my previous comment.